src/main/java/org/skriptlang/skript/bukkit/toolcomponent/elements/EffToolRuleDrops.java

```diff
@@ -0,0 +1,62 @@
+package org.skriptlang.skript.bukkit.toolcomponent.elements;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.*;
+import ch.njol.skript.lang.Effect;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+import org.bukkit.event.Event;
+import org.bukkit.inventory.meta.components.ToolComponent.ToolRule;
+import org.jetbrains.annotations.Nullable;
+import org.skriptlang.skript.bukkit.toolcomponent.ToolRuleWrapper;
+import org.skriptlang.skript.bukkit.toolcomponent.elements.EffSecCreateToolRule.ToolRuleEvent;
+
+@Name("Tool Rule - Drops")
+@Description("If the tool with this tool rule should drop items of a block defined from the blocks of this tool rule.")
+@Examples({
+	"create a new tool rule and store it in {_toolrule}:",
+		"\tset the tool rule blocks to oak log, stone and obsidian",
+		"\tset the tool rule speed to 5",
+		"\tenable tool rule drops",
+	"add {_toolrule} to the tool rules of {_item}"
+})
+@RequiredPlugins("Minecraft 1.20.6+")
+@Since("INSERT VERSION")
+public class EffToolRuleDrops extends Effect {
+
+	static {
+		Skript.registerEffect(EffToolRuleDrops.class,
+			"enable [the] tool rule drops [(of|for) %toolrule%]",
+			"disable [the] tool rule drops [(of|for) %toolrule%]");
+	}
+
+	private boolean enable;
+	private Expression<? extends ToolRule> exprToolRule;
+
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		if (!getParser().isCurrentEvent(ToolRuleEvent.class)) {
```

Comment:
What would happen if we tried changing the state outside of the section then? The rule should be modifiable no?
Especially since we're able to get references outside of it.